cpp/include/cudf/aggregation.hpp

```diff
@@ -159,6 +161,18 @@ std::unique_ptr<Base> make_sum_of_squares_aggregation();
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_mean_aggregation();
 
+/**
+ * @brief Factory to create a M2 aggregation
+ *
+ * A M2 aggregation is groupwise sum of squares of differences from the group mean. That is:
+ *  `M2(group) = SUM((x - MEAN(group)) * (x - MEAN(group)), for all x in group)`.
+ *
+ * This aggregation produces the intermediate values that are used to compute variance and standard
+ * deviation in distributed computing.
```

Comment:
We shouldn't refer to "distributed computing" anywhere as that's not the only reason you'd use this. You could also use this if you're processing a huge amount of data in batches on a single GPU. 

```suggestion
 * A M2 aggregation is sum of squares of differences from the mean. That is:
 *  `M2 = SUM((x - MEAN) * (x - MEAN), for all x)`.
 *
 * This aggregation produces the intermediate values that are used to compute variance and standard
 * deviation across multiple discrete sets. See https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm for more detail. 
```